desktop/apps/auction2/actions.js

```diff
@@ -12,11 +42,112 @@ export function updateArtworks(artworks) {
   }
 }
 
-export function toggleListView(isListView) {
+export function updateAggregatedArtists(aggregatedArtists) {
   return {
-    type: TOGGLE_LIST_VIEW,
+    type: UPDATE_AGGREGATED_ARTISTS,
     payload: {
-      isListView
+      aggregatedArtists
+    }
+  }
+}
+
+export function updateAggregatedMediums(aggregatedMediums) {
+  return {
+    type: UPDATE_AGGREGATED_MEDIUMS,
+    payload: {
+      aggregatedMediums
+    }
+  }
+}
+
+export function updateArtistParams(artistId) {
+  return (dispatch) => {
+    dispatch(updateArtistId(artistId))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateArtistId(artistId) {
+  return {
+    type: UPDATE_ARTIST_ID,
+    payload: {
+      artistId
     }
   }
 }
+
+export function updateMediumParams(mediumId) {
+  return (dispatch) => {
+    dispatch(updateMediumId(mediumId))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateMediumId(mediumId) {
+  return {
+    type: UPDATE_MEDIUM_ID,
+    payload: {
+      mediumId
+    }
+  }
+}
+
+export function updateSort(sort) {
+  return (dispatch) => {
+    dispatch(updateSortParam(sort))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateSortParam(sort) {
+  return {
+    type: UPDATE_SORT,
+    payload: {
+      sort
+    }
+  }
+}
+
+export function updateEstimateDisplay(min, max) {
+  return {
+    type: UPDATE_ESTIMATE_DISPLAY,
+    payload: {
+      min,
+      max
+    }
+  }
+}
+
+export function updateEstimateRange(min, max) {
+  return (dispatch) => {
+    dispatch(updateEstimateRangeParams(min, max))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateEstimateRangeParams(min, max) {
+  return {
+    type: UPDATE_ESTIMATE_RANGE,
+    payload: {
+      min,
+      max
+    }
+  }
+}
+
+export function fetchArtworks() {
+  return (dispatch, getState) => {
+    const { auctionArtworks: { filterParams } } = getState()
+    return metaphysics({ query: filterQuery, variables: filterParams, req: { user: sd.CURRENT_USER } })
+      .then(({ filter_artworks }) => {
+        const aggregations = filter_artworks.aggregations
+        const artistAggregation = _.filter(aggregations, (agg) => { return agg.slice == 'ARTIST' })
+        const mediumAggregation = _.filter(aggregations, (agg) => { return agg.slice == 'MEDIUM' })
```

Comment:
`filter` is native in JS, and also note the `===` which should always be used :

```javascript 
const artistAggregation = aggregations.filter(agg => agg.slice === 'ARTIST') 
...
```

Also, don't need `return` or brackets with the `=>`, especially if its a one liner